ledgereth/objects.py

```diff
@@ -113,7 +131,69 @@ def to_dict(self) -> dict:
         return d
 
 
+class Type2Transaction(rlp.Serializable):
+    """An unsigned Type 2 transaction.
+
+    Format spec:
+
+    0x02 || rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, amount, data, access_list])
+
+    Note: access_list is not actually implemented yet.
+    """
+
+    fields = [
+        ("chain_id", big_endian_int),
+        ("nonce", big_endian_int),
+        ("max_priority_fee_per_gas", big_endian_int),
+        ("max_fee_per_gas", big_endian_int),
+        ("gas_limit", big_endian_int),
+        ("destination", Binary.fixed_length(20, allow_empty=True)),
+        ("amount", big_endian_int),
+        ("data", binary),
+        # TODO: This only supports empty lists and needs to be fixed for
+        # EIP-2930 support
+        ("access_list", ListSedes()),
+    ]
+    transaction_type = TransactionType.EIP_1559
+
+    def __init__(
+        self,
+        chain_id: int,
+        nonce: int,
+        max_priority_fee_per_gas: int,
+        max_fee_per_gas: int,
+        gas_limit: int,
+        destination: bytes,
+        amount: int,
+        data: bytes,
+        # TODO: Access lists, type 1 transactions
+        access_list: List[bytes] = list(),
+    ):
+        super().__init__(
+            chain_id,
+            nonce,
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+            gas_limit,
+            destination,
+            amount,
+            data,
+            access_list,
+        )
+
+    def sender(self, value: str) -> None:
```

Comment:
Should have a more method like name?
It reminds me of a property-setter that way it is now, except it won't work that way:

https://python-reference.readthedocs.io/en/latest/docs/property/setter.html

Reply:
Curious, this appears to be a vestigial artifact from some before time and has no reference or use.  I'll just yank it.